packages/paste-website/src/components/shortcodes/table-of-contents/styles.ts

```diff
@@ -0,0 +1,63 @@
+import Scrollspy from 'react-scrollspy';
+import styled from '@emotion/styled';
+import {themeGet} from 'styled-system';
+import {Box} from '@twilio-paste/box';
+import {TableOfContentsListItemProps} from './types';
+
+export const StyledWrapper = styled(Box)`
+  order: 2;
+`;
+
+export const StyledScrollSpy = styled(Scrollspy)`
+  position: sticky;
+  top: 0;
+  right: 0;
+  margin: 0;
+  padding: 0;
+  list-style: none;
+`;
+
+export const marginTopPartial = ({depth}: TableOfContentsListItemProps): string => {
+  if (depth === '3' || depth === '4') {
+    return '0';
+  }
+  return themeGet('space.space30');
+};
+
+export const marginLeftPartial = ({depth}: TableOfContentsListItemProps): string => {
+  if (depth === '3' || depth === '4') {
+    return themeGet('space.space50');
+  }
+  return '0';
+};
+
+export const fontSizePartial = ({depth}: TableOfContentsListItemProps): string => {
+  if (depth === '3' || depth === '4') {
+    return themeGet('fontSizes.fontSize10');
+  }
+  return themeGet('fontSizes.fontSize20');
+};
+
+export const StyledListItem: React.FC<TableOfContentsListItemProps> = styled.li`
+  display: block;
+  margin-top: ${marginTopPartial};
+  margin-left: ${marginLeftPartial};
+  padding-left: ${themeGet('space.space30')};
+  font-size: ${fontSizePartial};
+  color: ${themeGet('colors.colorGray80')};
+  border-left-width: 2px;
+  border-left-style: solid;
+  border-left-color: transparent;
+
+  a {
+    color: ${themeGet('colors.colorGray80')};
+  }
+
+  &.is-current {
```

Comment:
This is kind of grim, and it's a fault of the library. Generally, components modifying their children is not a good idea. 

We don't have a choice here, but I'm just voicing my ick 😉

Reply:
As long as it is the exception - and isolated - I guess it's safe enough.  And I totally agree haha